### PR TITLE
[GH 12439] Convert `components/settings_sidebar` to TypeScript

### PR DESCRIPTION
**Pull request by wyze**
#### Summary

Convert `components/settings_sidebar` to TypeScript

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/12439

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -4909,7 +4924,6 @@
             "json5": "^2.1.0",
             "lodash": "^4.17.13",
             "resolve": "^1.3.2",
-            "semver": "^5.4.1",
```
Comment (devinbinnie) [inline package-lock.json]: Can we remove some of these unnecessary `package-lock.json` changes? We should only need the ones for the new types.

Comment (crspeller) [reply package-lock.json]: The package-lock.json file is automatic. If we modify it from the generated version it's just going to reset it isn't it?

Comment (wyze) [reply package-lock.json]: Done.

> The package-lock.json file is automatic. If we modify it from the generated version it's just going to reset it isn't it?

I usually use yarn so I may not be correct, but using different versions of `npm` when modifying the lock might generate different changes. I know with `yarn` sometimes I see integrity hashes being added for packages that remained unchanged.

**Comment by hanzei** [comment]
@wyze would you please merge `master` into your branch?

**Comment by wyze** [comment]
@hanzei Done!

**Pull request merged**